apps/fishing-map/features/track-correction/track-correction.slice.ts

```diff
@@ -56,12 +57,91 @@ const initialState: TrackCorrectionState = {
     },
     type: 'falsePositive',
   },
+  comment: '',
   issues: {
     status: AsyncReducerStatus.Idle,
     data: [],
   },
 }
 
+type CreateNewIssueThunkParam = {
+  issueBody: TrackCorrection
+  commentBody: TrackCorrectionComment
+  workspaceId: string
+}
+export const createNewIssueThunk = createAsyncThunk(
+  'trackCorrection/newIssue',
+  async (
+    { issueBody, commentBody, workspaceId }: CreateNewIssueThunkParam,
+    { rejectWithValue, dispatch }
+  ) => {
+    try {
+      const response = await fetch(`${PATH_BASENAME}/api/track-corrections/${workspaceId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          issueBody,
+          commentBody,
+        }),
+      })
+
+      if (!response.ok) {
+        const errorData = await response.json()
+        throw new Error(errorData.message || 'Failed to submit track correction')
+      }
+
+      const data = await response.json()
+
+      dispatch(resetTrackCorrection())
+
+      return data
+    } catch (e: any) {
+      return rejectWithValue(e.message || 'An unknown error occurred')
+    }
+  }
+)
+
+type CreateCommentThunkParam = {
+  issueId: string
+  commentBody: TrackCorrectionComment
+  workspaceId: string
+}
+export const createCommentThunk = createAsyncThunk(
+  'trackCorrection/comment',
+  async (
+    { issueId, commentBody, workspaceId }: CreateCommentThunkParam,
+    { rejectWithValue, dispatch }
+  ) => {
+    try {
+      const response = await fetch(`${PATH_BASENAME}/api/track-corrections/${workspaceId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          issueId,
+          commentBody,
+        }),
+      })
+
+      if (!response.ok) {
+        const errorData = await response.json()
+        throw new Error(errorData.message || 'Failed to submit track correction')
+      }
+
+      const data = await response.json()
+
+      dispatch(resetTrackCorrection())
```

Comment:
In general, I think dispatching other actions within actions can lead you to limit reusing the actions and coupling the code. What about moving the `dispatch(resetTrackCorrection())` where [it is invoked](https://github.com/GlobalFishingWatch/frontend/blob/ui-and-spreadsheet/apps/fishing-map/features/track-correction/TrackCorrection.tsx#L165) and use await to dispatch from there once finished without errors?
This would also allows us to reset only when needed, not every time